`dandi/tests/test_files.py::test_validate_bogus - assert False` on `=dandi-cli-0.55.0` and `=dandi-cli-0.55.1`

One test fails for some reason. This is the full build log → https://ppb.chymera.eu/46db45.log

Any ideas what could be wrong?

@jwodder 

```
decohost /var/tmp/portage/dev-vcs/dandi-cli-0.55.1/work/dandi-0.55.1 # DANDI_TESTS_NONETWORK=1 pytest -vvs dandi/tests/test_files.py
=========================================================== test session starts ============================================================
platform linux -- Python 3.11.3, pytest-7.3.1, pluggy-1.0.0 -- /usr/bin/python3.11
cachedir: .pytest_cache
rootdir: /var/tmp/portage/dev-vcs/dandi-cli-0.55.1/work/dandi-0.55.1
configfile: tox.ini
plugins: mock-3.10.0, rerunfailures-11.1.2, pkgcore-0.12.21, pyfakefs-5.2.2
collected 14 items

dandi/tests/test_files.py::test_find_dandi_files PASSED
dandi/tests/test_files.py::test_find_dandi_files_with_bids PASSED
dandi/tests/test_files.py::test_dandi_file_zarr_with_excluded_dotfiles PASSED
dandi/tests/test_files.py::test_validate_simple1 PASSED
dandi/tests/test_files.py::test_validate_simple1_no_subject PASSED
dandi/tests/test_files.py::test_validate_simple2 PASSED
dandi/tests/test_files.py::test_validate_simple2_new PASSED
dandi/tests/test_files.py::test_validate_simple3_no_subject_id PASSED
dandi/tests/test_files.py::test_validate_bogus ['Unable to synchronously open file (file signature not found)']
FAILED
dandi/tests/test_files.py::test_upload_zarr SKIPPED (no network settings)
dandi/tests/test_files.py::test_zarr_properties PASSED
dandi/tests/test_files.py::test_upload_zarr_with_excluded_dotfiles SKIPPED (no network settings)
dandi/tests/test_files.py::test_validate_deep_zarr PASSED
dandi/tests/test_files.py::test_validate_zarr_deep_via_excluded_dotfiles PASSED

================================================================= FAILURES =================================================================
___________________________________________________________ test_validate_bogus ____________________________________________________________
dandi/tests/test_files.py:378: in test_validate_bogus
    assert any(e.message.startswith("Unable to open file") for e in errors)
E   assert False
E    +  where False = any(<generator object test_validate_bogus.<locals>.<genexpr> at 0x7f8d47d399a0>)
=========================================================== slowest 10 durations ===========================================================
2.01s setup    dandi/tests/test_files.py::test_validate_simple2
0.09s call     dandi/tests/test_files.py::test_validate_simple2
0.08s call     dandi/tests/test_files.py::test_validate_simple1
0.08s call     dandi/tests/test_files.py::test_validate_simple2_new
0.08s call     dandi/tests/test_files.py::test_validate_simple1_no_subject
0.07s call     dandi/tests/test_files.py::test_validate_simple3_no_subject_id
0.03s setup    dandi/tests/test_files.py::test_validate_simple3_no_subject_id
0.03s setup    dandi/tests/test_files.py::test_validate_simple1
0.02s call     dandi/tests/test_files.py::test_zarr_properties
0.01s call     dandi/tests/test_files.py::test_validate_bogus
========================================================= short test summary info ==========================================================
FAILED dandi/tests/test_files.py::test_validate_bogus - assert False
================================================= 1 failed, 11 passed, 2 skipped in 2.73s ==================================================
```

Can also confirm this continues to happen on `dandi-0.55.1`

--- a/dandi/tests/test_files.py
+++ b/dandi/tests/test_files.py
@@ -374,7 +374,14 @@ def test_validate_bogus(tmp_path):
     errors = dandi_file(path).get_validation_errors()
     # ATM we would get 2 errors -- since could not be open in two places,
     # but that would be too rigid to test. Let's just see that we have expected errors
-    assert any(e.message.startswith("Unable to open file") for e in errors)
+    assert any(
+        e.message.startswith(
+            ("Unable to open file", "Unable to synchronously open file")
+        )
+        for e in errors
+    )
+    # Recent versions of hdf5 changed the error message, hence the need to
+    # check for two different patterns.
 
 
 def test_upload_zarr(new_dandiset, tmp_path):

